src/dash-table/derived/table/fragments.tsx

```diff
@@ -43,13 +59,32 @@ export default memoizeOneFactory((
                 },
                 { cells: 0, count: 0 },
                 row as any
-            ).cells),
-            cells) :
+            ).cells;
+
+            const res = row.slice(0, pivot).map((c,i) => getFixedColSpan(c, fixedColumns - i - 1)).concat(row.slice(pivot).map(getHiddenCell));
+            res[pivot - 1] = getLastOfType(res[pivot - 1]);
+
+            return res;
+        }, cells) :
         null;
 
-    cells = R.addIndex<JSX.Element[]>(R.map)((row, i) => row.slice(
-        (fixedColumnCells && fixedColumnCells[i].length) || 0
-    ), cells);
+    cells = isNil(fixedColumnCells) ?
+        cells :
+        R.map(row => {
+            const pivot = R.reduceWhile<JSX.Element, IAccumulator>(
+                acc => acc.count < fixedColumns,
+                (acc, cell) => {
+                    acc.cells++;
+                    acc.count += (cell.props.colSpan || 1);
+
+                    return acc;
+                },
+                { cells: 0, count: 0 },
+                row as any
+            ).cells;
```

Comment:
🌴`const pivot = ...` looks identical to above?

Reply:
🌴 https://github.com/plotly/dash-table/pull/722/commits/3d0b8c87f5828a7c6395b2209281a1571e6e36d6